megatron/model/transformer.py

```diff
@@ -259,6 +259,7 @@ def __init__(
             self.rotary_emb = None
 
         self.attention_type = neox_args.attention_config[layer_number]
+        self.use_flash_attention = self.attention_type == "flash"
         self.sparse = self.attention_type != "global"
         if self.sparse:
```

Comment:
I'm running into a bug here. Because `self.attention_type == "flash"`, `self.sparse == True` and I get the following stack trace when it tries to configure sparse attention. Did you not run into this issue @VHellendoorn ?

```
Traceback (most recent call last):
  File "/fsx/dashiell/gpt-neox/train.py", line 27, in <module>
    pretrain(neox_args=neox_args)
  File "/fsx/dashiell/gpt-neox/megatron/training.py", line 85, in pretrain
    model, optimizer, lr_scheduler = setup_model_and_optimizer(
  File "/fsx/dashiell/gpt-neox/megatron/training.py", line 403, in setup_model_and_optimizer
    model = get_model(neox_args=neox_args, use_cache=use_cache)
  File "/fsx/dashiell/gpt-neox/megatron/training.py", line 237, in get_model
    model = GPT2ModelPipe(
  File "/fsx/dashiell/gpt-neox/megatron/model/gpt2_model.py", line 123, in __init__
    super().__init__(
  File "/fsx/dashiell/DeeperSpeed/deepspeed/runtime/pipe/module.py", line 195, in __init__
    self._build()
  File "/fsx/dashiell/DeeperSpeed/deepspeed/runtime/pipe/module.py", line 246, in _build
    module = layer.build()
  File "/fsx/dashiell/DeeperSpeed/deepspeed/runtime/pipe/module.py", line 69, in build
    return self.typename(*self.module_args, **self.module_kwargs)
  File "/fsx/dashiell/gpt-neox/megatron/model/transformer.py", line 596, in __init__
    self.attention = ParallelSelfAttention(
  File "/fsx/dashiell/gpt-neox/megatron/model/transformer.py", line 265, in __init__
    self.sparse_attn = configure_sparse_attention(
  File "/fsx/dashiell/gpt-neox/megatron/model/utils.py", line 288, in configure_sparse_attention
    raise ValueError(f"Attention type {attention_type} not recognized")
ValueError: Attention type flash not recognized
```

Reply:
Continuing to test in branch [test_flash](https://github.com/EleutherAI/gpt-neox/tree/test_flash) changing this line to: 
`self.sparse = self.attention_type not in ("global", "flash")`

until I hear back from @VHellendoorn

Reply:
Ah you're right! My bad, I previously had the conditions here set up as `if flash: ... elif sparse: ... else: `, but realized this led to some code duplication. I must've benchmarked the version before this reordering. Your fix looks good; I'll push a commit with it in a second.